pkg/repositories/database.go

```diff
@@ -70,21 +77,24 @@ func getPostgresDsn(ctx context.Context, pgConfig runtimeInterfaces.PostgresConf
 		pgConfig.Host, pgConfig.Port, pgConfig.DbName, pgConfig.User, password, pgConfig.ExtraOptions)
 }
 
+// GetDB uses the dbConfig to create gorm DB object. If the db doesn't exist for the dbConfig then a new one is created
+// using the default db for the provider. eg : postgres has default dbName as postgres
+// TODO : support default db names across different providers.
 func GetDB(ctx context.Context, dbConfig *runtimeInterfaces.DbConfig, logConfig *logger.Config) (
 	*gorm.DB, error) {
 	if dbConfig == nil {
 		panic("Cannot initialize database repository from empty db config")
 	}
-	var dialector gorm.Dialector
 	logLevel := getGormLogLevel(ctx, logConfig)
 
+	var pgConfig runtimeInterfaces.PostgresConfig
```

Comment:
no postgres specific code outside the switch statement please!

Reply:
Agree. removed it.